node/mock-test.ts

```diff
@@ -125,4 +129,176 @@ export class MockTestRunner {
+        Object.keys(execution).forEach((key: string) => {
+            if (key.toLowerCase() == 'node') {
+                nodeVersion = 6;
+            }
+            else if (key.toLowerCase() == 'node10') {
+                nodeVersion = 10;
+            }
+        });
+        if (nodeVersion == 0) {
+            console.warn('Unable to determine execution type from task.json, defaulting to use Node 10');
+            return 10;
+        }
+
+        return nodeVersion;
+    }
+
+    // Returns the path to the task.json for the task being tested. Returns null if unable to find it.
+    // Searches by moving up the directory structure from the initial starting point and checking at each level.
+    private getTaskJsonPath(): string {
+        if (this._taskJsonPath) {
+            return this._taskJsonPath;
+        }
+        let curPath: string = this._testPath;
+        let newPath: string = path.join(this._testPath, '..');
+        while (curPath != newPath) {
+            curPath = newPath;
+            let taskJsonPath: string = path.join(curPath, 'task.json');
+            if (fs.existsSync(taskJsonPath)) {
+                return taskJsonPath;
+            }
+            newPath = path.join(curPath, '..');
+        }
+        return null;
+    }
+
+    // Downloads the specified node version to the download destination. Returns a path to node.exe
+    private downloadNode(nodeVersion: string, downloadDestination: string): string {
+        const nodeUrl: string = 'https://nodejs.org/dist';
+        switch (this.getPlatform()) {
+            case 'darwin':
+                this.downloadTarGz(nodeUrl + '/' + nodeVersion + '/node-' + nodeVersion + '-darwin-x64.tar.gz', downloadDestination);
+                return path.join(downloadDestination, 'node-' + nodeVersion + '-darwin-x64', 'bin', 'node');
+            case 'linux':
+                this.downloadTarGz(nodeUrl + '/' + nodeVersion + '/node-' + nodeVersion + '-linux-x64.tar.gz', downloadDestination);
+                return path.join(downloadDestination, 'node-' + nodeVersion + '-linux-x64', 'bin', 'node');
+            case 'win32':
+                this.downloadFile(nodeUrl + '/' + nodeVersion + '/win-x64/node.exe', downloadDestination, 'node.exe');
+                this.downloadFile(nodeUrl + '/' + nodeVersion + '/win-x64/node.lib', downloadDestination, 'node.lib');
+                return path.join(downloadDestination, 'node.exe')
+        }
+    }
+
+    //Downloads file to the downloadDestination, making any necessary folders along the way.
+    private downloadFile(url: string, downloadDestination: string, fileName: string): void {
+        if (!url) {
+            throw new Error('Parameter "url" must be set.');
+        }
+        if (!downloadDestination) {
+            throw new Error('Parameter "downloadDestination" must be set.');
+        }
+        console.log('Downloading file:', url);
+        shelljs.mkdir('-p', downloadDestination);
```

Comment:
should we just use fs? move away from npm install warnings

Reply:
I think I like leaving this for now because `fs.mkdirSync` doesn't have a recursive option in Node<10. I can make that, but I don't feel a need to do that until we decide we're completely getting rid of the dependency.

Reply:
I'm fine either way.

Note, mkdirP is in task lib. do we have access to that function here?

Reply:
Not without import `task.ts` which would probably be fine, but I'm a little more inclined to leave this for now and handle it (along with the rest of shelljs) in a future PR.